feature/login/src/main/java/com/epmedu/animeal/login/code/domain/EnterCodeViewModel.kt

```diff
@@ -0,0 +1,93 @@
+package com.epmedu.animeal.login.code.domain
+
+import androidx.lifecycle.viewModelScope
+import com.epmedu.animeal.common.domain.StateViewModel
+import com.epmedu.animeal.login.code.data.EnterCodeRepository
+import com.epmedu.animeal.login.code.domain.model.EnterCodeState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+internal class EnterCodeViewModel @Inject constructor(
+    private val repository: EnterCodeRepository
+) : StateViewModel<EnterCodeState>(EnterCodeState()) {
+
+    private val _isCodeCorrect = MutableStateFlow(false)
+    val isCodeCorrect: StateFlow<Boolean> = _isCodeCorrect.asStateFlow()
+
+    init {
+        viewModelScope.launch {
```

Comment:
launch inside launch?)

maybe better `viewModelScope.launch{}` and `viewModelScope.launch {}`?

Reply:
Ok)